### ISSUE TITLE
Chronos seems to have stuck values for chronosBindAverage and chronosRegAverage stats

### ISSUE THREAD

**Issue by rkd-msw** [Metaswitch/chronos#328]
#### Symptoms

I registered 1,200 subscribers with a 30-minute expiry time at 12:00. By 12:45, those should have all expired, but my Chronos stats still report nonzero averages and instantaneous counts:

```
root@abra:~# snmpwalk -m PROJECT-CLEARWATER-MIB -v2c -c clearwater stress-sprout-2.stress.clearwater.test PROJECT-CLEARWATER-MIB::chronosBindAverage
PROJECT-CLEARWATER-MIB::chronosBindAverage.scopePrevious5SecondPeriod = Gauge32: 54
PROJECT-CLEARWATER-MIB::chronosBindAverage.scopeCurrent5MinutePeriod = Gauge32: 21
PROJECT-CLEARWATER-MIB::chronosBindAverage.scopePrevious5MinutePeriod = Gauge32: 21
root@abra:~# snmpwalk -m PROJECT-CLEARWATER-MIB -v2c -c clearwater stress-sprout-2.stress.clearwater.test PROJECT-CLEARWATER-MIB::chronosRegAverage
PROJECT-CLEARWATER-MIB::chronosRegAverage.scopePrevious5SecondPeriod = Gauge32: 4
PROJECT-CLEARWATER-MIB::chronosRegAverage.scopeCurrent5MinutePeriod = Gauge32: 5
PROJECT-CLEARWATER-MIB::chronosRegAverage.scopePrevious5MinutePeriod = Gauge32: 5
root@abra:~# snmpwalk -m PROJECT-CLEARWATER-MIB -v2c -c clearwater stress-sprout-2.stress.clearwater.test PROJECT-CLEARWATER-MIB::chronosRegInstantaneousCount
PROJECT-CLEARWATER-MIB::chronosRegInstantaneousCount.0 = Gauge32: 10
root@abra:~# snmpwalk -m PROJECT-CLEARWATER-MIB -v2c -c clearwater stress-sprout-2.stress.clearwater.test PROJECT-CLEARWATER-MIB::chronosBindInstantaneousCount
PROJECT-CLEARWATER-MIB::chronosBindInstantaneousCount.0 = Gauge32: 7
```

#### Impact

Users may overestimate the usage of their deployment (potentially creating larger deployments than necessary, or being reluctant to have maintenance windows as their deployment seems to be permanently in use).

#### Release and environment 

`10.0.0-161216-~255.00.0-161216.020317`

This deployment has been up for a little over 10 days.

#### Steps to reproduce

Unknown - specifically, I don't know whether just running stress once against a clean deployment will consistently reproduce this, although I'll try and find out.

**Comment by rkd-msw** [comment]
Diags packages from both Sprout nodes at `\\ad\csd\bigfiles\clearwater\issue-attachments\chronos\328`.

**Comment by rkd-msw** [comment]
I restarted both Chronos processes, checked all the stats were 0, then ran one stress run. This reproduces, so it looks like it is easily reproducible:

```
root@abra:~# date; snmpwalk -m PROJECT-CLEARWATER-MIB -v2c -c clearwater stress-sprout-1.stress.clearwater.test PROJECT-CLEARWATER-MIB::chronos
Wed  4 Jan 15:05:23 GMT 2017
PROJECT-CLEARWATER-MIB::chronosRemainingNodesQuery.0 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosTimersProcessedCount.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosTimersProcessedCount.2 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosTimersProcessedCount.3 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosInvalidTimersProcessedCount.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosInvalidTimersProcessedCount.2 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosInvalidTimersProcessedCount.3 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersAverage.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersAverage.scopeCurrent5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersAverage.scopePrevious5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersVariance.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersVariance.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersVariance.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersHWM.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersHWM.scopeCurrent5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersHWM.scopePrevious5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersLWM.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersLWM.scopeCurrent5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersLWM.scopePrevious5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersActions.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersActions.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersActions.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallInstantaneousCount.0 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubInstantaneousCount.0 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindInstantaneousCount.0 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegInstantaneousCount.0 = Gauge32: 1
PROJECT-CLEARWATER-MIB::chronosCallAverage.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallAverage.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallAverage.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallVariance.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallVariance.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallVariance.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallHWM.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallHWM.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallHWM.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallLWM.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallLWM.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallLWM.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubAverage.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubAverage.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubAverage.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubVariance.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubVariance.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubVariance.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubHWM.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubHWM.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubHWM.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubLWM.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubLWM.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubLWM.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindAverage.scopePrevious5SecondPeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosBindAverage.scopeCurrent5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosBindAverage.scopePrevious5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosBindVariance.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindVariance.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindVariance.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindHWM.scopePrevious5SecondPeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosBindHWM.scopeCurrent5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosBindHWM.scopePrevious5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosBindLWM.scopePrevious5SecondPeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosBindLWM.scopeCurrent5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosBindLWM.scopePrevious5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosRegAverage.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegAverage.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegAverage.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegVariance.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegVariance.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegVariance.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegHWM.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegHWM.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegHWM.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegLWM.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegLWM.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegLWM.scopePrevious5MinutePeriod = Gauge32: 0
root@abra:~# date; snmpwalk -m PROJECT-CLEARWATER-MIB -v2c -c clearwater stress-sprout-2.stress.clearwater.test PROJECT-CLEARWATER-MIB::chronos
Wed  4 Jan 15:05:25 GMT 2017
PROJECT-CLEARWATER-MIB::chronosRemainingNodesQuery.0 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosTimersProcessedCount.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosTimersProcessedCount.2 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosTimersProcessedCount.3 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosInvalidTimersProcessedCount.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosInvalidTimersProcessedCount.2 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosInvalidTimersProcessedCount.3 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersAverage.scopePrevious5SecondPeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersAverage.scopeCurrent5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersAverage.scopePrevious5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersVariance.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersVariance.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersVariance.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersHWM.scopePrevious5SecondPeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersHWM.scopeCurrent5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersHWM.scopePrevious5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersLWM.scopePrevious5SecondPeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersLWM.scopeCurrent5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersLWM.scopePrevious5MinutePeriod = Gauge32: 2
PROJECT-CLEARWATER-MIB::chronosAllTimersActions.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersActions.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosAllTimersActions.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallInstantaneousCount.0 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubInstantaneousCount.0 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindInstantaneousCount.0 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegInstantaneousCount.0 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallAverage.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallAverage.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallAverage.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallVariance.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallVariance.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallVariance.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallHWM.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallHWM.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallHWM.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallLWM.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallLWM.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosCallLWM.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubAverage.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubAverage.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubAverage.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubVariance.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubVariance.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubVariance.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubHWM.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubHWM.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubHWM.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubLWM.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubLWM.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosSubLWM.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindAverage.scopePrevious5SecondPeriod = Gauge32: 1
PROJECT-CLEARWATER-MIB::chronosBindAverage.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindAverage.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindVariance.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindVariance.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindVariance.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindHWM.scopePrevious5SecondPeriod = Gauge32: 1
PROJECT-CLEARWATER-MIB::chronosBindHWM.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindHWM.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindLWM.scopePrevious5SecondPeriod = Gauge32: 1
PROJECT-CLEARWATER-MIB::chronosBindLWM.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosBindLWM.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegAverage.scopePrevious5SecondPeriod = Gauge32: 1
PROJECT-CLEARWATER-MIB::chronosRegAverage.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegAverage.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegVariance.scopePrevious5SecondPeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegVariance.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegVariance.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegHWM.scopePrevious5SecondPeriod = Gauge32: 1
PROJECT-CLEARWATER-MIB::chronosRegHWM.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegHWM.scopePrevious5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegLWM.scopePrevious5SecondPeriod = Gauge32: 1
PROJECT-CLEARWATER-MIB::chronosRegLWM.scopeCurrent5MinutePeriod = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegLWM.scopePrevious5MinutePeriod = Gauge32: 0
```

**Comment by johadalin** [comment]
Should be resolved by #599 . Passing back for re-test.

**Comment by rkd-msw** [comment]
Fixed in 10.0.0-170118-~255.00.0-170118.020317:

```
rkd@abra:~/salinas$ date;snmpwalk -m PROJECT-CLEARWATER-MIB -v2c -c clearwater stress-sprout-1.stress.clearwater.test PROJECT-CLEARWATER-MIB::chronosRegInstantaneousCount.0; snmpwalk -m PROJECT-CLEARWATER-MIB -v2c -c clearwater stress-sprout-2.stress.clearwater.test PROJECT-CLEARWATER-MIB::chronosRegInstantaneousCount.0
Sat 21 Jan 11:13:54 GMT 2017
PROJECT-CLEARWATER-MIB::chronosRegInstantaneousCount.0 = Gauge32: 74969
PROJECT-CLEARWATER-MIB::chronosRegInstantaneousCount.0 = Gauge32: 74969
rkd@abra:~/salinas$ date;snmpwalk -m PROJECT-CLEARWATER-MIB -v2c -c clearwater stress-sprout-1.stress.clearwater.test PROJECT-CLEARWATER-MIB::chronosRegInstantaneousCount.0; snmpwalk -m PROJECT-CLEARWATER-MIB -v2c -c clearwater stress-sprout-2.stress.clearwater.test PROJECT-CLEARWATER-MIB::chronosRegInstantaneousCount.0
Sat 21 Jan 13:21:03 GMT 2017
PROJECT-CLEARWATER-MIB::chronosRegInstantaneousCount.0 = Gauge32: 0
PROJECT-CLEARWATER-MIB::chronosRegInstantaneousCount.0 = Gauge32: 0
```
